tests/test_modeling_tf_encoder_decoder.py

```diff
@@ -0,0 +1,871 @@
+            "decoder_token_labels": decoder_token_labels,
+            "decoder_choice_labels": decoder_choice_labels,
+            "encoder_hidden_states": encoder_hidden_states,
+            "labels": decoder_token_labels,
+        }
+
+
+@require_tf
+class TFEncoderDecoderModelTest(unittest.TestCase):
+    def get_from_encoderdecoder_pretrained_model(self):
+        return TFEncoderDecoderModel.from_encoder_decoder_pretrained("bert-base-cased", "bert-base-cased")
+
+    def get_decoder_config(self):
+        config = AutoConfig.from_pretrained("bert-base-cased")
+        config.is_decoder = True
+        config.add_cross_attention = True
+        return config
+
+    def get_encoderdecoder_model(self):
+        return TFEncoderDecoderModel.from_pretrained("patrickvonplaten/bert2bert-cnn_dailymail-fp16", from_pt=True)
+
+    def get_encoder_decoder_models(self):
+        encoder_model = TFBertModel.from_pretrained("bert-base-cased", name="encoder")
+        decoder_model = TFBertLMHeadModel.from_pretrained(
+            "bert-base-cased", config=self.get_decoder_config(), name="decoder"
+        )
+        return {"encoder": encoder_model, "decoder": decoder_model}
+
+    def _check_configuration_tie(self, model):
+        assert id(model.decoder.config) == id(model.config.decoder)
+        assert id(model.encoder.config) == id(model.config.encoder)
+
+    @slow
+    def test_configuration_tie(self):
+        model = self.get_from_encoderdecoder_pretrained_model()
+        self._check_configuration_tie(model)
+
+        model = TFEncoderDecoderModel(**self.get_encoder_decoder_models())
+        self._check_configuration_tie(model)
+
+        model = self.get_encoderdecoder_model()
+        self._check_configuration_tie(model)
+
+
+@require_tf
+class TFEncoderDecoderModelSaveLoadTests(unittest.TestCase):
+    def get_encoder_decoder_config(self):
+        encoder_config = AutoConfig.from_pretrained("bert-base-uncased")
+        decoder_config = AutoConfig.from_pretrained("bert-base-uncased", is_decoder=True, add_cross_attention=True)
+        return EncoderDecoderConfig.from_encoder_decoder_configs(encoder_config, decoder_config)
+
+    def get_encoder_decoder_config_small(self):
+        encoder_config = AutoConfig.from_pretrained("julien-c/bert-xsmall-dummy")
+        decoder_config = AutoConfig.from_pretrained(
+            "julien-c/bert-xsmall-dummy", is_decoder=True, add_cross_attention=True
+        )
+        return EncoderDecoderConfig.from_encoder_decoder_configs(encoder_config, decoder_config)
+
+    def test_encoder_decoder_save_load_from_encoder_decoder(self):
+        config = self.get_encoder_decoder_config_small()
+
+        # create two random BERT models for bert2bert & initialize weights (+cross_attention weights)
+        encoder = TFBertModel(config.encoder)
+        encoder(encoder.dummy_inputs)
+        decoder = TFBertLMHeadModel(config.decoder)
+        decoder(decoder.dummy_inputs)
```

Comment:
Hey, @patrickvonplaten I saw this test passed! Great.

I felt a bit confused, because I had the same test before, but having issue about the weights not loading correctly.

After checking your commit here, I found the issue I had previously is because I didn't have:

```
encoder(encoder.dummy_inputs)
decoder(decoder.dummy_inputs)
```

Great you have this test working now :), thank you!

Reply:
I will check tomorrow if the issue from `load_pytorch_model_in_tf2_model` called in `test_pt_tf_model_equivalence` can be fixed by doing similar things (i.e. make sure tf model is built before loading weights)